editoast/src/views/timetable.rs

```diff
@@ -442,23 +446,84 @@ async fn conflicts(
     )
     .await?;
 
+    let mut paced_trains_ts = vec![];
+    for paced_train in paced_trains.clone() {
+        if paced_train.step.num_seconds() == 0 {
+            paced_trains_ts.push(paced_train.into_first_occurrence());
+            continue;
+        }
+
+        let occurrences = paced_train.duration.num_seconds() / paced_train.step.num_seconds();
+        let step = paced_train.step;
+        let mut start_time = paced_train.start_time;
+
+        let all_occurrences: Vec<TrainSchedule> = (0..occurrences)
+            .map(|_| {
+                let mut first_occurrence = paced_train.clone().into_first_occurrence();
+                first_occurrence.start_time = start_time;
+                start_time += step;
+                first_occurrence
+            })
+            .collect();
```

Comment:
Weird style, either go fully functionnal with `map`/`reduce` or fully imperative, it's very unusual to have a `map` with side-effects.